src/net-mdns/src/MulticastClient.cs

```diff
@@ -0,0 +1,254 @@
+﻿using Common.Logging;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+
+namespace Makaretu.Dns
+{
+    /// <summary>
+    ///   Performs the magic to send and receive datagrams over multicast
+    ///   sockets.
+    /// </summary>
+    class MulticastClient : IDisposable
+    {
+        static readonly ILog log = LogManager.GetLogger(typeof(MulticastClient));
+
+        /// <summary>
+        ///   The port number assigned to Multicast DNS.
+        /// </summary>
+        /// <value>
+        ///   Port number 5353.
+        /// </value>
+        public static readonly int MulticastPort = 5353;
+
+        static readonly IPAddress MulticastAddressIp4 = IPAddress.Parse("224.0.0.251");
+        static readonly IPAddress MulticastAddressIp6 = IPAddress.Parse("FF02::FB");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make sure using this hardcoded IP address 'FF02::FB' is safe here.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)